src/module-elasticsuite-core/Helper/Autocomplete.php

```diff
@@ -51,6 +51,68 @@ public function isEnabled($type)
         return $this->getMaxSize($type) > 0;
     }
 
+    /**
+     * Check if Autocomplete "extension" system is enabled.
+     *
+     * @return bool
+     */
+    public function isExtensionEnabled()
+    {
+        return (bool) $this->scopeConfig->isSetFlag(
+            self::AUTOCOMPLETE_SETTINGS_CONFIG_XML_PREFIX . "/advanced/extension_enabled",
+            \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+        );
+    }
+
+    /**
+     * Check if Autocomplete "extension" system is limited.
+     *
+     * @return bool
+     */
+    public function isExtensionLimited()
+    {
+        return (bool) $this->scopeConfig->isSetFlag(
+            self::AUTOCOMPLETE_SETTINGS_CONFIG_XML_PREFIX . "/advanced/extension_limit",
+            \Magento\Store\Model\ScopeInterface::SCOPE_STORE
+        );
```

Comment:
If you want to allow a value of "0" for the extension_limit, ie, "No limit", then it should be 
```
isSetFlag && getExtensionSize() > 0
```

Reply:
Nope, the "no limit" is the case where **"extension_limit = false".**